src/pow.cpp

```diff
@@ -134,9 +134,24 @@ bool PermittedDifficultyTransition(const Consensus::Params& params, int64_t heig
     return true;
 }
 
+/*
+// Bypasses the actual proof of work check during fuzz testing with a dummy validation checking whether nNonce is odd or even.
+bool CheckProofOfWork(uint256 hash, unsigned int nBits, uint32_t nNonce, const Consensus::Params& params)
+{
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+    return nNonce & 1;
+#else
+    return CheckProofOfWorkImpl(hash, nBits, params);
+#endif
+}
+*/
+
 bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params& params)
 {
-    bool fNegative;
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
```

Comment:
We still need a separate `CheckProofOfWorkImpl` that can be fuzzed on its own.

Otherwise, https://github.com/bitcoin/bitcoin/blob/5fdbc8b4ee609d7f58dd4dff30e506a7400cd768/src/test/fuzz/pow.cpp#L83 would not be testing anything.

Reply:
done